### PR TITLE
Torxakis blueprint: tester

### PR DESCRIPTION
**Pull request by dnadales**
Preliminary steps towards blueprinting a *IOCO tester (IOCO, UIOCO, etc). At this stage we're not concerned about data (as far as I can see), and this code defines the scaffolding of what I think it could be a convenient architecture for this aspect of the tool.

Please review the ideas proposed here. If they don't make sense or are too naive it is always possible to cancel the pull request. But remember we're at the blueprinting stage anyway.

CC: @HarmMunkESI

### CODE REVIEW

**Comment by capitanbatata** [comment]
The sample console runner is not aware of the current terminal colors, but it should look like this in a black background:

screenshot from 2017-12-14 19-06-45

**Comment by capitanbatata** [comment]
And this is how the current dependencies look like. See also the README!

torxakis-intra-deps

File: sys/txs-core/src/Test/TorXakis/Examples/ConsoleRunner.hs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    return $ SeqChecker expsTQ passTV retuTV
+
+instance SUTConnection SeqChecker where
+    startSUT = const $ return ()
+
+    fromSUT SeqChecker{passing, returnOutput} = 
+        ifM (readTVarIO returnOutput) sendOutput waitForever
+        where 
+          sendOutput = atomically $ do
+              writeTVar returnOutput False -- We need to wait for an input before this function can be called again.
```
Comment (pjljvandelaar) [inline sys/txs-core/src/Test/TorXakis/Examples/ConsoleRunner.hs]: Disagree: multiple outputs can occur without the need for input!

Comment (capitanbatata) [reply sys/txs-core/src/Test/TorXakis/Examples/ConsoleRunner.hs]: Remember that this is an example. A seq chequer does not send more than one output. Other instances might allow that of course.

**Comment by pjljvandelaar** [comment]
I need a voice over....

File: docs/torxakis-intra-deps.svg [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (keremispirli) [inline docs/torxakis-intra-deps.svg]: If this file is generated, then we can omit it from the repository

File: sys/txs-core/src/Test/TorXakis/Action.hs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+data Action = Action { actionName :: Text }
+            | Quiescence
+    deriving (Eq)
+
+instance Show Action where
+    show (Action name) = show name
+    show Quiescence = "Quiescence"
+
+instance IsString Action where
+    fromString = Action . T.pack -- Yes, yes, for now we could have Action "Quiescence".
```
Comment (keremispirli) [inline sys/txs-core/src/Test/TorXakis/Action.hs]: "For now" means, until when?

Comment (capitanbatata) [reply sys/txs-core/src/Test/TorXakis/Action.hs]: Errrr until we take the next step in this blueprint?

Comment (keremispirli) [reply sys/txs-core/src/Test/TorXakis/Action.hs]: OK :)

**Comment by dnadales** [comment]
Just cleaning up mi github PR's.

**Pull request closed without merging**